perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -3891,6 +3883,37 @@ private void storeApplicationAttributes(Application app) throws UserNotExistsExc
 
 	}
 
+	/**
+	 * Normalize input value from registration form for map values and correctly merge them with the originalValue from User.
+	 *
+	 * @param originalValue Value from Users attribute currently stored in Perun
+	 * @param newValue Value provided by registration form
+	 * @return Map of attribute keys, values after merge
+	 */
+	private LinkedHashMap<String, String> handleMapValue(LinkedHashMap<String, String> originalValue, String newValue) {
+
+		// Normalize value representing map
+		String preparedVal = newValue.replaceAll("(\n)+", ",");
```

Comment:
I would be careful with this first normalization of input value. It was originally done to handle list values from multi-line input fields. After all these changes, we probably shouldn't use such widgets to handle list attributes, then it shouldn't be necessary to normalize this and would actually malform multi-line input.
I'm not saying it shouldn't be there, just we should know all possible use-cases and constraint for list/map attribute values and make changes and safeguards accordingly. It should be all tested thoroughly.

Reply:
I agree that this normalization of data will probably not be needed with the new widgets and so decided to remove it. However as You said, some sort of processing may be needed as a outcome of the implementation and testing of WUI part of this task.